nook/functions/doc_translator/translation_orchestrator.py

```
@@ -0,0 +1,431 @@
+                    # Update progress if monitor is available
+                    if self.monitor:
+                        self._update_progress(len(results), len(documents), "translation")
+
+                except Exception as e:
+                    if self.error_recovery_enabled:
+                        # Attempt recovery
+                        logger.warning(f"Translation failed, attempting recovery: {e}")
+                        try:
+                            result = self.translation_engine.translate_document_section(doc.content)
+                            results.append(result)
+                        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)